vaadin-split-layout-flow-parent/vaadin-split-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/splitlayout/tests/SplitterPositionIT.java

```diff
@@ -102,6 +102,31 @@ public void nestedSplitLayouts_moveParentSplitter_getSplitterPosition() {
         Assert.assertTrue(splitPosition < 81);
     }
 
+    @Test
+    public void setSplitterPositionFromServer_moveOnClient_resetToOriginal() {
+        // Add split layout with 30% splitter position
+        $(NativeButtonElement.class).id("createLayoutJavaApi").click();
+        $(NativeButtonElement.class).id("setSplitPositionJavaApi").click();
+
+        var split = $(SplitLayoutElement.class).id("splitLayoutJavaApi");
+        var primaryComponent = split.getPrimaryComponent();
+
+        var flexBasisInitial = primaryComponent.getCssValue("flex-basis");
+
+        // Move splitter by 150px
+        var splitter = split.getSplitter();
+        Actions resizeAction = new Actions(getDriver());
+        resizeAction.dragAndDropBy(splitter, 150, 0);
+        resizeAction.perform();
+
```

Comment:
Does it make sense to also check whether `flex-basis` has been properly updated? I looked at the other tests and there doesn't seem to be any for this. We can maybe add something like `Assert.assertNotEquals(flexBasisInitial, primaryComponent.getCssValue("flex-basis"));` or add another test to be on the safe side. It is not a blocker though. What do you think?